### ISSUE TITLE
Multiplayer

### ISSUE THREAD

**Issue by kermitdafrog8** [d07RiV/diabloweb#99]
Multiplayer says game exist when creating but doesn't exist when trying to connect.

**Comment by dpastern** [comment]
Same issue here.

**Comment by St4tikk** [comment]
I'm having the same problem.   :(

**Comment by sskras** [comment]
Seems like @d07RiV just suspended both the development and hosting. I would say, You have to wait for another developer and probably hoster.

**Comment by kermitdafrog8** [comment]
It's still up but still unable to connect to multiplayer.  I would actually like to host it locally for myself on my server.

**Comment by Knotschi** [comment]
I've also added a PR #113  for this change in case the @d07RiV is not planning to bring his server back up

**Comment by kermitdafrog8** [comment]
So it's possible to actually play with more than 1 player?

**Comment by Knotschi** [comment]
It worked for me fine using my local network. Through the internet it lagged a lot for me, most of the time it was not even possible to join a game. But maybe the internet connection was just not good enough. I am not sure.
It also works fine on the same computer with different browsers (e.g. for exchanging gear between chars)

I also realized that sometimes, when it was not working anymore I had to restart the game (reload the browser) and than it was working again.

please try it out and let me know if it works for you

**Comment by kermitdafrog8** [comment]
Is there any way to contact you directly?

**Comment by Knotschi** [comment]
you can email me (my github username @posteo.de )

**Comment by sskras** [comment]
@kermitdafrog8 wrote:
> It's still up but still unable to connect to multiplayer. I would actually like to host it locally for myself on my server.

By stopped hosting I meant WebRTC, the multiplayer service. The files are present, yes, that's why this repo is receiving reports about failing multiplayer mode.

Going to try the @Knotschi fork.

**Comment by kermitdafrog8** [comment]
I have been on his fork playing single player on multiplayer mode.

**Comment by nakele** [comment]
Yep, just tried his and it's working. Tested with 2 devices (mobile and desktop), 2 players WebRTC. You need to have your own MPQ otherwise it fails. I have the full one but I'm assuming the shareware might work as well.

**Comment by kermitdafrog8** [comment]
It works if I use Firefox and Chrome on the same computer.

**Comment by nakele** [comment]
AH, gotcha. It only works if the devices are within the same subnetwork. AKA all connected to the same wifi.

**Comment by kermitdafrog8** [comment]
So you can't play against someone else outside your own network?

**Comment by nakele** [comment]
You might be able with a Virtual network. Could try "Logmein" or make your own VPN server + client.

**Comment by AJenbo** [comment]
Having to both run a server and VPN seams to deafat the point of webdiablo, woudn't using DevilutionX make more sens at that point? There are still several unsolved issues in webdiablo and if you are on Android/iOS maybe the time would be better spend finishing the native ports for thoes platforms as you would get a better interface and widescreen support.
https://github.com/diasurgical/devilutionX

**Comment by Knotschi** [comment]
WebRTC should actually also work the same way over the internet as inside a local network. I am also not sure if a VPN would help here. I could imagine that the problem is just related to the higher latency over the web. But this would need to be tested.
If I find some time for it I would like to do some further digging and maybe also try out to get the websocket version running but currently I have not time for that.

**Comment by cojmar** [comment]
you need a signaling server if u wanna use webRTC over internet,
https://www.webrtc-experiment.com/docs/WebRTC-Signaling-Concepts.html

**Comment by sskras** [comment]
@Knotschi I have some issue playing the Shareware version on your fork.
Should it be reported?

I see [knotschi/diabloweb](https://github.com/knotschi/diabloweb) doesn't allow creating issues in the repo.

**Comment by Knotschi** [comment]
@cojmar Yes, this is why the version from @d07RiV wasn't working anymore, he shut down his server which was used as the signaling server. In my version I am using the default server from https://peerjs.com/ (which is a js library for webRTC which was used, it provides a default signaling server that you can use instead of setting up your own)
This signaling server is not only used when you use webRTC over internet, also for creating connections in the local network to the computers can find each other.

@sskras you are right, I haven't tried that. I will try to fix it later. I've also enabled issues on my repo now

**Comment by tiehr** [comment]
It's been a long time and I still can't play multiplayer. I'll be waiting

**Comment by Knotschi** [comment]
@tiehr You mean over the internet? At least local network multiplayer works now fine for me on https://knotschi.github.io/diabloweb/

**Comment by edenLOL** [comment]
@tiehr I know it's been a while, but if it helps, I successfully played Knotschi's fork with 3 other friends over the internet (no VPN/Hamachi service etc) yesterday. This was using Chrome with the full MPQ.

Thanks for your fix @Knotschi , I've had a blast so far!

**Comment by nakele** [comment]
> @tiehr I know it's been a while, but if it helps, I successfully played Knotschi's fork with 3 other friends over the internet (no VPN/Hamachi service etc) yesterday. This was using Chrome with the full MPQ.
> 
> Thanks for your fix @Knotschi , I've had a blast so far!

What configuration did u use? Windows? Wifi? Phone hotspot? What os? What device? What internet provider? Thank you :)

**Comment by edenLOL** [comment]
> What configuration did u use? Windows? Wifi? Phone hotspot? What os? What device? What internet provider? Thank you :)

OS: Windows 10 on desktop PC
Connection type: ethernet (but my friends were on WiFi at their houses)

All of us are on stable fibre connections in Australia/New Zealand.

One of us hosted using WebRTC peer (or whichever the top option is in the multiplayer), and the others simply entered the game name. The time to connect was about 5-10 seconds.

**Comment by nakele** [comment]
> > What configuration did u use? Windows? Wifi? Phone hotspot? What os? What device? What internet provider? Thank you :)
> 
> OS: Windows 10 on desktop PC
> Connection type: ethernet (but my friends were on WiFi at their houses)
> 
> All of us are on stable fibre connections in Australia/New Zealand.
> 
> One of us hosted using WebRTC peer (or whichever the top option is in the multiplayer), and the others simply entered the game name. The time to connect was about 5-10 seconds.

Thank you  very much,  will try again. My suspicious now is that phone network providers block something . . .

**Comment by edenLOL** [comment]
> > > What configuration did u use? Windows? Wifi? Phone hotspot? What os? What device? What internet provider? Thank you :)
> > 
> > 
> > OS: Windows 10 on desktop PC
> > Connection type: ethernet (but my friends were on WiFi at their houses)
> > All of us are on stable fibre connections in Australia/New Zealand.
> > One of us hosted using WebRTC peer (or whichever the top option is in the multiplayer), and the others simply entered the game name. The time to connect was about 5-10 seconds.
> 
> Thank you very much, will try again. My suspicious now is that phone network providers block something . . .

Out of curiosity, I just tried to connect between my Macbook and my Android mobile (both using Chrome). It worked on WiFi, but failed (joined, then timed out) when my phone was on the mobile network (off WiFi). 

I think a lot of providers (especially mobile network) block port 443 that the public PeerJS server relies on.

**Comment by nakele** [comment]
> > > > What configuration did u use? Windows? Wifi? Phone hotspot? What os? What device? What internet provider? Thank you :)
> > > 
> > > 
> > > OS: Windows 10 on desktop PC
> > > Connection type: ethernet (but my friends were on WiFi at their houses)
> > > All of us are on stable fibre connections in Australia/New Zealand.
> > > One of us hosted using WebRTC peer (or whichever the top option is in the multiplayer), and the others simply entered the game name. The time to connect was about 5-10 seconds.
> > 
> > 
> > Thank you very much, will try again. My suspicious now is that phone network providers block something . . .
> 
> Out of curiosity, I just tried to connect between my Macbook and my Android mobile (both using Chrome). It worked on WiFi, but failed (joined, then timed out) when my phone was on the mobile network (off WiFi).
> 
> I think a lot of providers (especially mobile network) block port 443 that the public PeerJS server relies on.

Same, pc on wifi + android on mobile network

**Comment by imneckro** [comment]
Guess im little late but ill give it a shot if any kind soul has the time to help me out.
When im selecting multiplayer it gives me the following error:
![image](https://user-images.githubusercontent.com/42493772/178810207-5368b4bc-8924-4b23-b9c9-91b9b256c63f.png)
Does anyone know what i have to configure on my server end?
Many thanks in advance!
@eden-lecomte maybe you can show me the right way to do it?

**Comment by AJenbo** [comment]
No clue, have you tried DevilutionX instead?

**Comment by imneckro** [comment]
> No clue, have you tried DevilutionX instead?

Didnt try it. Does DevilutionX build for web?

**Comment by AJenbo** [comment]
It does build for web, but is currently not support. But it is native for most platforms, what are you trying to run it on/needing it for?

**Comment by imneckro** [comment]
@AJenbo im hosting this on apache server. I want to test multiplayer support but when selecting multiplayer gives me:
![178810207-5368b4bc-8924-4b23-b9c9-91b9b256c63f](https://user-images.githubusercontent.com/42493772/179361512-9df78da7-6cfa-45ec-999e-4b002397edf4.png)

I read older messages saying that multiplayer was working.

**Comment by nakele** [comment]
![Screenshot_20220716-164252_Firefox Nightly](https://user-images.githubusercontent.com/12185661/179361900-e3fcb01d-942a-4e54-b28e-082c113b1b9e.jpg)
> @AJenbo im hosting this on apache server. I want to test multiplayer support but when selecting multiplayer gives me: ![178810207-5368b4bc-8924-4b23-b9c9-91b9b256c63f](https://user-images.githubusercontent.com/42493772/179361512-9df78da7-6cfa-45ec-999e-4b002397edf4.png)
> 
> I read older messages saying that multiplayer was working.

Use this link and then direct option for multipley, one create and the other joins: https://knotschi.github.io/diabloweb/

Tested now on my phone on 2 separate browsers

Edit: Apologies I don't think I understood that you wanted to run it on ur own server.

**Comment by imneckro** [comment]
@AJenbo I’ve made some personal changes in d07RiV source. So I m building it myself. I can’t figure out Knotschis differences to d07RiV and why Knotschis multiplayer is functional while d07RiV gives “No Network selected”.

**Comment by kermitdafrog8** [comment]
It sounded like it was something with WebRTC hosting.

**Comment by imneckro** [comment]
@kermitdafrog8 did you spot Knotschi's differences vs d07RiV?

**Comment by kermitdafrog8** [comment]
@imneckro did you look at https://github.com/d07RiV/diabloweb/pull/113?  This might be the answer to your question.  Not sure.

**Comment by imneckro** [comment]
@kermitdafrog8 I end up building the forked one from Kotschis repo and all works fine.

**Comment by imneckro** [comment]
I’m hosting it through Apache server. But multiplayer giving the error I attached in picture.  https://knotschi.github.io/diabloweb/  runs multiplayer correct.
I can’t figure out knotschi’s differences.



Sent from myMail for iOS


Thursday, 14 July 2022, 02:24 +0300 from ***@***.***  ***@***.***>:

**Comment by Knotschi** [comment]
The small change I made was already merged: https://github.com/d07RiV/diabloweb/pull/113
So my fork should not be needed anymore.

But it seems that this one here has a different issue now which leads to the error you show.

I think this commit might have introduced the new issue “No Network selected": https://github.com/d07RiV/diabloweb/commit/998629fd5d9f87ba9693cf1409cf2acea603c06e
